MekHQ/src/mekhq/campaign/io/CampaignXmlParser.java

```diff
@@ -581,11 +583,59 @@ public Campaign parse() throws CampaignXmlParseException, NullEntityException {
                         }
                     });
 
+
+
+        if (MHQConstants.VERSION.isHigherThan(version)) {
+            triggerActiveContractWarning(retVal);
+        }
+
         logger.info("Load of campaign file complete!");
 
         return retVal;
     }
 
+    /**
+     * Displays a warning dialog if the given {@code Campaign} has an active or future contract.
+     * <p>
+     * This method checks the campaign's list of contracts to determine if there are any currently active
+     * or accepted contracts with a start date in the future. If such a contract exists, it shows an error
+     * dialog informing the user that the campaign cannot proceed until the contract is completed.
+     * </p>
+     *
+     * @param campaign The {@link Campaign} object representing the current campaign,
+     *               which contains information about the contracts and local date.
+     */
+    private static void triggerActiveContractWarning(Campaign campaign) {
+        boolean hasActiveContract = !campaign.getActiveAtBContracts().isEmpty();
+
+        if (!hasActiveContract) {
+            List<AtBContract> contracts = campaign.getAtBContracts();
+            LocalDate today = campaign.getLocalDate();
+
+            for (AtBContract contract : contracts) {
+                // This catches any contracts that have been accepted, but haven't yet started
+                if (contract.getStartDate().isAfter(today)) {
+                    hasActiveContract = true;
+                    break;
+                }
+            }
+        }
+
+        if (hasActiveContract) {
+            String message = String.format(
+                "This campaign has an active contract." +
+                    "\n\nPlease complete the contract before updating to version %s.",
+                MHQConstants.VERSION);
+
+            JOptionPane.showMessageDialog(
+                null,
+                message,
+                "Error",
+                JOptionPane.ERROR_MESSAGE
+            );
+        }
```

Comment:
I believe there are two variables here, `hasActiveContract` and `hasFutureContract`
then the message dialog should then show `if (hasActiveContract(campaign) || hasFutureContract(campaign))`

Make both variables static methods that return a boolean, this will make the calculation of the `hasFutureContract` be conditioned by the first, so if there is no current active contract it will check if there is a future active contract. The two new static functions are small and could be used for many things in the future.